source/1_engine/rex_engine/src/system_info.cpp

```diff
@@ -0,0 +1,136 @@
+#include "rex_engine/system_info.h"
+
+namespace rex
+{
+  //-------------------------------------------------------------------------
+  SystemInfo::SystemInfo()
+  {
+#if defined REX_PLATFORM_WINDOWS
```

Comment:
I think it'd be cleaner to merge as many of these ifdefs as possible
#if defined REX_PLATFORM_WINDOWS
m_platform = platform::Type::Windows
#if defined REX_API_OPENGL
m_graphics_api = graphics_api::Type::OpenGL
...